File: app-server/src/main/java/com/github/arhor/simple/expense/tracker/data/repository/ExpenseRepository.java
package com.github.arhor.simple.expense.tracker.data.repository;

import java.util.List;

import org.springframework.data.repository.CrudRepository;

import com.github.arhor.simple.expense.tracker.data.model.Expense;

public interface ExpenseRepository extends CrudRepository<Expense, Long> {

    List<Expense> findAllByUserId(Long userId);
}

File: app-server/src/main/kotlin/com/github/arhor/simple/expense/tracker/data/repository/ExpenseRepository.kt
package com.github.arhor.simple.expense.tracker.data.repository

import com.github.arhor.simple.expense.tracker.data.model.Expense
import org.springframework.data.repository.CrudRepository

interface ExpenseRepository : CrudRepository<Expense, Long> {

    fun findAllByUserId(userId: Long): List<Expense>
}
